faq.md

```diff
@@ -152,19 +165,31 @@ if the refresh token is not valid anymore, drop the session (and navigate to the
 Simple code example:
 
 ```dart
-interceptors: [
-  // Auth Interceptor
-  (Request request) async => applyHeader(request, 'authorization',
-      SharedPrefs.localStorage.getString(tokenHeader),
-      override: false),
-  (Response response) async {
+class AuthInterceptor implements Interceptor {
+
+  @override
+  FutureOr<Response<BodyType>> intercept<BodyType>(Chain<BodyType> chain) async {
+    final request = applyHeader(chain.request, 'authorization',
+        SharedPrefs.localStorage.getString(tokenHeader),
+        override: false);
+   
+    final response = await chain.proceed(request);
+   
     if (response?.statusCode == 401) {
       SharedPrefs.localStorage.remove(tokenHeader);
       // Navigate to some login page or just request new token
```

Comment:
How does one break out of the interceptor if there's no way to refresh the session? 

Right now, there's nothing to indicate to the reader whether they should throw an error if they can't attempt a refresh anymore. Should they throw an Error or return `null` here?

Consider making this clearer so that a reader knows what to do in this situation.

Reply:
Here I just updated the example to reflect the refactor changes. But I do agree that its not fully clear. While I don't think it should be added here. 

I will include it in the Interceptor docs that its possible to break out by throwing a exception.

Reply:
I think it should be included in both so that it's readily available when perusing through the Markdown docs.

Please lemme know what you think.